sdk/textanalytics/Azure.AI.TextAnalytics.Protocol/README.md

```diff
@@ -12,12 +12,307 @@ At this time, if you are looking to interact with the Azure Cognitive Services T
+In addition, you can use `SetEmptyObject` and `SetEmptyArray` to add a property with an empty array or object as a value. These methods return the newly created value, so you can modify it:
+
+```csharp
+var doc = new JsonData(); // represents the document: {}
+var wrapped = doc.SetEmptyObject("wrapped"); // doc now represents the document { "wrapped": { } }
+wrapped["message"] = "Hello, JsonData!"; // doc now represents the document { "wrapped": { "message": "This works, too!" } }
+```
+
+**Note**: The type of the indexer property is `JsonData`. While there are implicit conversions from primitive CLR values and `string` to `JsonData`, there isn't one for objects, so you may need to explicitly call `new JsonData(<value>)` when assigning to the indexer property.
+
+##### Modifying Arrays
+
+To modify an array, use the indexer:
+
+```csharp
+var doc = new JsonData(new[] { "Hello, JsonData!" }); // represents the document [ "Hello, JsonData!" ]
+doc[0] = "This works!"; // represents the document [ "This works!" ]
+```
+
+To add a new value to the end of the array, use the `Add` method:
+
+```csharp
+var doc = new JsonData(new[] {}); // represents the document [ ]
+doc.Add("This works!"); // represents the document [ "This works!" ]
+```
+
+There are also helper methods for adding an empty array or object to the end of this array:
+
+```csharp
+var doc = new JsonData(new[] {}); // represents the document: [ ]
+var wrapped = doc.AddEmptyObject(); // doc now represents the document [ { } ]
+wrapped["message"] = "Hello, JsonData!"; // doc now represents the document [ { "message": "This works, too!" } ]
+```
+
+**Note**: The type of the indexer property is `JsonData`. While there are implicit conversions from primitive CLR values and `string` to `JsonData`, there isn't one for objects, so you may need to explicitly call `new JsonData(<value>)` when assigning to the indexer property.
+
+#### Reading JSON Documents
+
+There are multiple ways to read data from a `JsonData`
+
+##### Using DOM APIs
+
+If a `JsonData` represents a primitive value or a string, you can cast to the coresponding CLR type:
+
+```csharp
+var oneDoc = new JsonData(1);           // represents the document: 1
+int oneValue = (int)oneDoc;             // works, oneValue is 1
+
+var stringDoc = new JsonData("hello");  // represents the document: "hello"
+int stringValue = (string)stringDoc;     // works, stringValue is the string "hello"
+```
+
+If the `JsonData` represents an object, you can use the `Properties` property to list all the properties of an object:
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" }); // represents the document: { "key1": "one", "key2": "two" }
+Console.WriteLine(string.Join(',' objectDoc.Properties)); // prints "key1, key2"
+```
+
+And you can fetch a specific property by name, using the indexer:
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" });
+Console.WriteLine(objectDoc["key1"]); // prints "one"
```

Comment:
I wonder if all these samples that get values should not include casts. Otherwise, I think people will think that the main way to access values is through ToString.

Reply:
I will update the examples here to use casts.  I do think that having `ToString` do "the right thing" for primitives is super useful, epically when it comes to string interpolation.  It will be interesting to see what users try naturally.

Reply:
I added an explicit section about how ToString() works, but did not modify the samples where we are just passing stuff to Console.WriteLine for now.  I think the casts make everything read so much worse.  For cases where we assign to a specific string variable, I did include the cast.